test/relayHub/RelayHubContractScenarios.test.ts

```diff
@@ -0,0 +1,737 @@
+
+      await assert.isRejected(
+        mockRelayHub.connect(relayWorker).relayCall(relayRequest, signature),
+        ERR_UNSTAKED,
+        'Relay request was properly processed'
+      );
+    });
+
+  });
+
+  describe('Manager - DeployRequest scenarios', function() {
+    const url = 'http://relay.com';
+
+    const HARDHAT_CHAIN_ID = 31337;
+    let token: FakeContract;
+    let smartWallet: FakeContract<SmartWallet>;
+    let externalWallet: Wallet;
+    const nextWalletIndex = 500;
+
+    function createRequest(
+      request: Partial<IForwarder.DeployRequestStruct>,
+      relayData: Partial<RelayData>
+    ): DeployRequest {
+      const baseRequest: DeployRequest = {
+        request: {
+          relayHub: ZERO_ADDRESS,
+          from: ZERO_ADDRESS,
+          to: ZERO_ADDRESS,
+          tokenContract: ZERO_ADDRESS,
+          value: '0',
+          nonce: '0',
+          tokenAmount: '1',
+          tokenGas: '50000',
+          data: '0x',
+          recoverer: '0',
+          index: 0,
+        },
+        relayData: {
+          gasPrice: '1',
+          relayWorker: ZERO_ADDRESS,
+          callForwarder: ZERO_ADDRESS,
+          callVerifier: ZERO_ADDRESS
+        }
+      }
+
+      return {
+        request: {
+          ...baseRequest.request,
+          ...request,
+        },
+        relayData: {
+          ...baseRequest.relayData,
+          ...relayData,
+        },
+      };
+    }
+
+    beforeEach(async function() {
+      token = await smock.fake('ERC20');
+      token.transfer.returns(true);
+
+      deployRelayHubContract = createContractDeployer(fakePenalizer.address);
+      mockRelayHub = await deployRelayHubContract();
+
+      smartWallet = await smock.fake('SmartWallet');
+      externalWallet = ethers.Wallet.createRandom();
+      externalWallet.connect(ethers.provider);
+    });
+
+    afterEach(function () {
+      mockRelayHub = undefined as unknown as MockContract<RelayHub>;
+    });
+
+    it('Should failed a deployRequest if SmartWallet has already been initialized', async function() {
+
+      await mockRelayHub.stakeForAddress(relayManagerAddr, 1000, {
+        value: ethers.BigNumber.from(2),
+        from: relayOwnerAddr,
+      });
+
+      await mockRelayHub.connect(relayManager).addRelayWorkers([relayWorkerAddr]);
+      await mockRelayHub.connect(relayManager).registerRelayServer(url);
+
+      const smartWalletFactory: FakeContract<SmartWalletFactory> = await smock.fake<SmartWalletFactory>("SmartWalletFactory");
+      smartWalletFactory.relayedUserSmartWalletCreation.reverts();
+
+      const relayRequestMisbehavingVerifier = createRequest({
+        from: externalWallet.address,
+        to: ZERO_ADDRESS,
+        data: '0x',
+        tokenContract: token.address,
+        relayHub: mockRelayHub.address,
+        tokenAmount: '1',
+        tokenGas: '50000',
+        recoverer: ZERO_ADDRESS,
+        index: '0',
+        gas: '90000000000000'
```

Comment:
There is a type error here. I don't know the solution but I think it should be addressed. Same at line 684.

Reply:
Fixed with an extension of the type declaration. Suggested by @jurajpiar